backend/functions/media_event.ts

```diff
@@ -0,0 +1,63 @@
+import { failure, IEvent, success } from '../libs/response';
+import { findRoomById } from '../libs/room-operations';
+import { updateWatcherVideoStatus } from '../libs/watcher-operations';
+import { MediaEventForServer, Room } from './../../extension/src/types';
+
+const findAndEnsureRoomJoined = async (
+    roomId: string,
+    watcherConnectionString: string,
+): Promise<Room | undefined> => {
+    if (!process.env.ROOM_TABLE) {
+        throw new Error('env.ROOM_TABLE must be defined');
+    }
+
+    if (!roomId) {
+        console.log(
+            '[WS-S] Could not find an existing roomId in the join room request',
+        );
+        return undefined;
+    }
+    const room = await findRoomById(roomId, process.env.ROOM_TABLE);
+
+    if (!room) {
+        console.log('[WS-S] Could not find room =_=');
+        return undefined;
+    }
+    if (!room.watchers[watcherConnectionString]) {
+        console.log(
+            '[WS-S] A media event was received for someone ho has not joined the room. Dropping',
+        );
+        return undefined;
+    }
+    return room;
+};
+
+export const main = async (event: IEvent) => {
+    const mediaEvent = JSON.parse(event.body) as MediaEventForServer;
+    const { roomId, playerEvent } = mediaEvent;
+    const watcherId = event.requestContext.connectionId;
+
+    const room = await findAndEnsureRoomJoined(roomId, watcherId);
```

Comment:
The functions does tzo things. It might be best to split it into two functions> One of them (EnsureRoomJoined can be written in the lib folder)